mainhub/mainhub.ino

```diff
@@ -114,6 +130,28 @@ void loop()
     draw();
     last_draw = millis();
   }
-
-  //  delay(10);
+  if (is_button_pressed) {
+    if (cursor_index >= min_index && cursor_index <= max_index) {
+        // Game won!
+      Serial.println("Game won!");
+      delay(1000);
+      cursor_index = 0;
+      target_width -= 1;
+      wait_update = wait_update/2;
```

Comment:
This looks great!

For managing future complexity, it might be helpful to have a variable `int current_level` which tracks which level you are on. 

that way you can derive all the other values such as the animations and speeds from it.